### PR TITLE
feat:implemented the working of buttons and seekbar in the waveform panel

### LINKED ISSUE #958: Implement the working of buttons in Waveform panel in Wave Generator
**Actual Behaviour**

Only the UI of waveform is there.

**Expected Behaviour**

All the buttons in the waveform panel should be working and connected to the monitors.

**Steps to reproduce it**

WaveGenerator - > Create the working of all buttons in waveform panel.

**Would you like to work on the issue?**

Yes

### PR DESCRIPTION
**Pull request by Avjeet**
Fixes #958 
Partly fixes the issue #878

Changes:

-  Implemented the working of the buttons present in the waveform part in the controlling panel in the Wave Generator. 
- The buttons are working and are connected to the Left Hand Side Monitor Screen which shows the properties of the waveforms.
- The seekbar is working and up and down buttons can be used to increase decrease value

Screenshots for the change: 
ezgif com-video-to-gif 1

APK for testing: 
[app-debug.zip](https://github.com/fossasia/pslab-android/files/2086761/app-debug.zip)

### CODE REVIEW

**Comment by harsh-2711** [comment]
@Avjeet  Please squash the commits

File: app/build.gradle [hunk 1]
```diff
@@ -50,6 +50,8 @@ dependencies {
     implementation "com.github.tiagohm.MarkdownView:library:$rootProject.markDownViewVersion"
     implementation "com.github.mirrajabi:search-dialog:$rootProject.searchDialogVersion"
     implementation "com.github.BeppiMenozzi:Knob:$rootProject.knobLibraryVersion"
+    implementation 'com.github.warkiz.widget:indicatorseekbar:1.2.9'
```
Comment (Vikum94) [inline app/build.gradle]: Version should be specified at the application gradle file. Check how other libraries have implemented.

File: app/src/main/java/org/fossasia/pslab/activity/WaveGeneratorActivity.java [hunk 2]
```diff
@@ -1,15 +1,359 @@
...
+        Double freqValue = (double)WaveGeneratorCommon.wave.get(btn_Active).get(FREQUENCY);
+
+        String freqValueText = freqValue + " " + getString(R.string.unit_hz);
+        waveFreqValue.setText(freqValueText);
+
+        Double value = (double)WaveGeneratorCommon.wave.get(btn_Active).get(PHASE);
+
+        String valueText = value + " " + getString(R.string.deg_text);
+        wavePhaseValue.setText(valueText);
+
```
Comment (abhinavraj23) [inline app/src/main/java/org/fossasia/pslab/activity/WaveGeneratorActivity.java]: @Avjeet Give this a code reformat, there are various useless spaces

File: app/src/main/res/values/colors.xml [hunk 3]
```diff
@@ -14,6 +14,8 @@
     <color name="white_grey">#9ea7aa</color>
     <color name="dark_grey">#757575</color>
     <color name="orange">#ff7043</color>
+    <color name="theme_red_light">#40c72c2c</color>
     <color name="blue_grey">#546E7A</color>
 
+
```
Comment (abhinavraj23) [inline app/src/main/res/values/colors.xml]: No need of this extra line

File: app/src/main/java/org/fossasia/pslab/others/WaveGeneratorCommon.java [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+        wave.get(WaveGeneratorActivity.WAVE1).put(WaveGeneratorActivity.FREQUENCY,10);
+
+        wave.get(WaveGeneratorActivity.WAVE2).put(WaveGeneratorActivity.PHASE,0);
+        wave.get(WaveGeneratorActivity.WAVE2).put(WaveGeneratorActivity.FREQUENCY,10);
+
+        wave.get(WaveGeneratorActivity.WAVE1).put(WaveGeneratorActivity.WAVETYPE,1);
+        wave.get(WaveGeneratorActivity.WAVE2).put(WaveGeneratorActivity.WAVETYPE,1);
+    }
+}
```
Comment (abhinavraj23) [inline app/src/main/java/org/fossasia/pslab/others/WaveGeneratorCommon.java]: Please give this a code reformat.

File: app/src/main/java/org/fossasia/pslab/activity/WaveGeneratorActivity.java [hunk 5]
```diff
@@ -1,15 +1,359 @@
...
+            @Override
+            public void onClick(View view) {
+                WaveGeneratorCommon.wave.get(btn_Active).put(WAVETYPE, SQUARE);
+                selectWaveform(SQUARE);
+            }
+        });
+        imgBtnTri.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                WaveGeneratorCommon.wave.get(btn_Active).put(WAVETYPE, SQUARE);
```
Comment (Vikum94) [inline app/src/main/java/org/fossasia/pslab/activity/WaveGeneratorActivity.java]: Is this supposed to be Triangular instead of Square?

Comment (Avjeet) [reply app/src/main/java/org/fossasia/pslab/activity/WaveGeneratorActivity.java]: I will make that square but changing the phase for wave1 was not there in first place in control module

**Comment by Avjeet** [comment]
@abhinavraj23 @Vikum94 @CloudyPadmal Plz review it and merge it as soon as possible

**Pull request merged**